pkgs/by-name/em/emacspeak/package.nix

```diff
@@ -1,34 +1,38 @@
 { lib
-, stdenv
+, emacs
+, espeak-ng
 , fetchFromGitHub
 , makeWrapper
-, emacs
+, stdenv
 , tcl
 , tclx
-, espeak-ng
 }:
 
-stdenv.mkDerivation rec {
+assert lib.versionAtLeast emacs.version "29";
```

Comment:
I think meta.broken would be better.

Reply:
I don't believe it is appropriate for the case here.

The [upstream NEWS](https://github.com/tvraman/emacspeak/blob/59.0/etc/NEWS) say Emacs 29.1 is required.
This is a different thing from saying "it has a bug that hinders it from running on older Emacsen".

Reply:
Broken does not merely mean that the build will not complete. It can also mean that the package won't work as expected at runtime, even if it builds fine.

Declaring a package as "broken" is the standard method of communicating that a package does not function. Please use that instead of a custom assertion.

Reply:
> Broken does not merely mean that the build will not complete.

This is irrelevant for my argument. Buildtime failures and runtime failures are both undesired outcomes.

There is a difference between "I didn't figure out how to make it work on Clang+Darwin+Arm" and "the upstream does not provide support for older Emacsen".

It makes no sense, e.g., to consider Linux 3.x as broken because it does not compile on [GCC < 4](https://www.kernel.org/doc/html/next/process/changes.html). `unsupported` would be a better name for that specific scenario.

Nonetheless, I will remove this commit.

Reply:
We have an RFC for more accurate package warnings: https://github.com/NixOS/rfcs/pull/127

Once that lands, you'd add a more specific optional problem to the package Though even then I'd argue its kind would be `broken` rather than `unsupported` as the latter implies a platform incompatibility.

Reply:
> It makes no sense, e.g., to consider Linux 3.x as broken because it does not compile on [GCC < 4](https://www.kernel.org/doc/html/next/process/changes.html).

If that was a likely configuration, yes, yes it would.

> Nonetheless, I will remove this commit.

That was not my intent. It's important to inform the user of breaking changes whenever possible.

Reply:
> We have an RFC for more accurate package warnings: https://github.com/NixOS/rfcs/pull/127

Good to know. I will watch this now.

> It's important to inform the user of breaking changes whenever possible.

I will add a comment explaining the broken status, then.